### PR TITLE
Feature/task 2

### PR DESCRIPTION
**Pull request by titoffag**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 
+import { CoursesListPageComponent } from './pages/courses-list-page/courses-list-page.component';
 
-const routes: Routes = [];
+
+const routes: Routes = [
+  { path: 'courses-list', component: CoursesListPageComponent },
+  { path: '**', redirectTo: 'courses-list' },
```
Comment (vitaliy-bobrov) [inline src/app/app-routing.module.ts]: I would just redirect from empty path - `{ path: '', redirectTo: 'courses-list',  pathMatch: 'full' }`

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [RouterModule.forRoot(
+    routes,
+    { enableTracing: true },
```
Comment (vitaliy-bobrov) [inline src/app/app-routing.module.ts]: nice to use environments to disable debug on prod builds

File: src/app/components/breadcrumbs/breadcrumbs.component.html [hunk 3]
```diff
@@ -1 +1,4 @@
-<p>breadcrumbs works!</p>
+<p-breadcrumb
+  [style]="{ padding: '10px 85px' }"
```
Comment (vitaliy-bobrov) [inline src/app/components/breadcrumbs/breadcrumbs.component.html]: could you omit style binding - try to use `ui-breadcrumb` for styling, see primeng styling docs

File: src/app/components/breadcrumbs/breadcrumbs.component.ts [hunk 4]
```diff
@@ -7,9 +8,14 @@ import { Component, OnInit } from '@angular/core';
 })
 export class BreadcrumbsComponent implements OnInit {
 
+  public items: MenuItem[];
```
Comment (vitaliy-bobrov) [inline src/app/components/breadcrumbs/breadcrumbs.component.ts]: public is optional as it is default

File: src/app/components/breadcrumbs/breadcrumbs.component.ts [hunk 5]
```diff
@@ -7,9 +8,14 @@ import { Component, OnInit } from '@angular/core';
 })
```
Comment (vitaliy-bobrov) [inline src/app/components/breadcrumbs/breadcrumbs.component.ts]: nice to have change detection onpush for dumb component - see https://blog.angularindepth.com/everything-you-need-to-know-about-change-detection-in-angular-8006c51d206f

File: src/app/components/breadcrumbs/breadcrumbs.component.ts [hunk 6]
```diff
@@ -7,9 +8,14 @@ import { Component, OnInit } from '@angular/core';
...
 export class BreadcrumbsComponent implements OnInit {
 
+  public items: MenuItem[];
+
   constructor() { }
```
Comment (vitaliy-bobrov) [inline src/app/components/breadcrumbs/breadcrumbs.component.ts]: remove if unused

File: src/app/components/course-item/course-item.component.html [hunk 7]
```diff
@@ -0,0 +1,32 @@
+<li>
+  <p-card [style]="{ margin: '25px 0' }">
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.html]: ditto

File: src/app/components/course-item/course-item.component.html [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    <p-header>
+      <h2 class="course-title">
+        {{course.title}}
+      </h2>
+      <span class="course-duration">
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.html]: nice to use time tag - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time

File: src/app/components/course-item/course-item.component.html [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        <fa-icon class="icon" [icon]="faClock"></fa-icon>
+        {{course.duration}} min
+      </span>
+      <span class="course-date">
+        <fa-icon class="icon" [icon]="faCalendarAlt"></fa-icon>
+        {{course.creationDate | date:'d MMM, y'}}
+      </span>
+    </p-header>
+
+    <div class="course-description">
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.html]: `<p>`

File: src/app/components/course-item/course-item.component.html [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+      {{course.description}}
+    </div>
+
+    <p-footer>
+      <p-button label="Edit" (onClick)="editCourse()"
+        [style]="{ backgroundColor: '#30b6dd', border: '1px solid #30b6dd', position: 'relative', paddingLeft: '20px' }">
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.html]: move to CSS if possible - as styles are not dynamic

File: src/app/components/course-item/course-item.component.ts [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+import { ICourse } from 'src/app/models/course';
+
+@Component({
+  selector: 'app-course-item',
+  templateUrl: './course-item.component.html',
+  styleUrls: ['./course-item.component.scss']
+})
+export class CourseItemComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() course: ICourse;
+  @Output() deletedCourse = new EventEmitter<number>();
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.ts]: why not to just emit id as string or change the course interface ?

File: src/app/components/course-item/course-item.component.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+  public faPencilAlt = faPencilAlt;
+  public faTrashAlt = faTrashAlt;
+  public faCalendarAlt = faCalendarAlt;
+  public faClock = faClock;
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.ts]: I would omit public keyword as it is default

File: src/app/components/courses-list/courses-list.component.html [hunk 13]
```diff
@@ -1 +1,8 @@
-<p>courses-list works!</p>
+<ul class="courses-list">
+  <app-course-item *ngFor="let course of courses" [course]="course" (deletedCourse)="onDeletedCourse($event)">
```
Comment (vitaliy-bobrov) [inline src/app/components/courses-list/courses-list.component.html]: nice to use trackBy - https://netbasal.com/angular-2-improve-performance-with-trackby-cc147b5104e5

File: src/app/components/courses-list/courses-list.component.html [hunk 14]
```diff
@@ -1 +1,8 @@
...
+  </app-course-item>
+</ul>
+<p-card styleClass="course-card load-more"
+  [style]="{ margin: '25px 0 100px', display: 'flex', justifyContent: 'center' }">
+  <a href="#" (click)="loadMoreCourses($event)">Load more</a>
```
Comment (vitaliy-bobrov) [inline src/app/components/courses-list/courses-list.component.html]: use `<button>` instead

File: src/app/components/courses-list/courses-list.component.ts [hunk 15]
```diff
@@ -1,15 +1,99 @@
...
+
+  ngAfterViewInit(): void {
+    console.log('[Courses List] ngAfterViewInit hook call');
+  }
+
+  ngAfterViewChecked(): void {
+    console.log('[Courses List] ngAfterViewChecked hook call');
+  }
+
+  loadMoreCourses(event: MouseEvent): void {
```
Comment (vitaliy-bobrov) [inline src/app/components/courses-list/courses-list.component.ts]: optional - I prefer to omit void when method / function doesn't return value

File: src/app/components/courses-list/courses-list.component.ts [hunk 16]
```diff
@@ -1,15 +1,99 @@
...
+    event.preventDefault();
+    console.log('load more courses');
+  }
+
+  onDeletedCourse(id: number): void {
+    console.log('course deleted with id:', id);
+    this.courses = this.courses.filter(course => {
+      return +course.id !== id;
+    });
```
Comment (vitaliy-bobrov) [inline src/app/components/courses-list/courses-list.component.ts]: `course => +course.id !== id;`

File: src/app/components/footer/footer.component.scss [hunk 17]
```diff
@@ -0,0 +1,15 @@
+@import 'src/styles.scss';
+
+.footer {
+  position: fixed;
```
Comment (vitaliy-bobrov) [inline src/app/components/footer/footer.component.scss]: you can try position sticky - https://developer.mozilla.org/en-US/docs/Web/CSS/position

File: src/app/components/logo/logo.component.scss [hunk 18]
```diff
@@ -0,0 +1,10 @@
+@import 'src/styles';
```
Comment (vitaliy-bobrov) [inline src/app/components/logo/logo.component.scss]: you you need this import?

File: src/app/components/logo/logo.component.scss [hunk 19]
```diff
@@ -0,0 +1,10 @@
...
+
+.logo {
```
Comment (vitaliy-bobrov) [inline src/app/components/logo/logo.component.scss]: nice to encapsulate styles using `:host` - https://blog.angular-university.io/angular-host-context/

File: src/app/components/logo/logo.component.ts [hunk 20]
```diff
@@ -7,6 +8,8 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LogoComponent implements OnInit {
 
+  public faPlayCircle = faPlayCircle;
+
   constructor() { }
 
   ngOnInit() {
```
Comment (vitaliy-bobrov) [inline src/app/components/logo/logo.component.ts]: remove if not used

File: src/app/components/panel/panel.component.html [hunk 21]
```diff
@@ -1 +1,14 @@
-<p>panel works!</p>
+<div class="panel-top">
+  <span class="ui-float-label search">
+    <input id="float-input" type="text" size="30" pInputText [(ngModel)]="searchInput">
```
Comment (vitaliy-bobrov) [inline src/app/components/panel/panel.component.html]: IMO `type="search"` would fit better

File: src/app/components/panel/panel.component.ts [hunk 22]
```diff
@@ -7,9 +8,17 @@ import { Component, OnInit } from '@angular/core';
 })
 export class PanelComponent implements OnInit {
 
+  public searchInput = '';
+  public faPlus = faPlus;
+
   constructor() { }
 
   ngOnInit() {
   }
```
Comment (vitaliy-bobrov) [inline src/app/components/panel/panel.component.ts]: ditto

File: src/app/components/user-profile/user-profile.component.html [hunk 23]
```diff
@@ -1 +1,8 @@
-<p>user-profile works!</p>
+<span>
+  <fa-icon [icon]="faUser"></fa-icon>
+  User login
+</span>
+<span class="user-profile">
+  <fa-icon [icon]="faSignOutAlt"></fa-icon>
+  <button class="log-off" type="submit">Log off</button>
```
Comment (vitaliy-bobrov) [inline src/app/components/user-profile/user-profile.component.html]: should it be type="submit"?

File: src/index.html [hunk 24]
```diff
@@ -8,6 +8,6 @@
   <link rel="icon" type="image/x-icon" href="favicon.ico">
 </head>
 <body>
-  <app-root></app-root>
+  <app-root class="app"></app-root>
```
Comment (vitaliy-bobrov) [inline src/index.html]: just use custom element tag for styling

File: src/app/components/breadcrumbs/breadcrumbs.component.ts (hunk 6 above)
Comment (titoffag) [reply src/app/components/breadcrumbs/breadcrumbs.component.ts]: I use this variable in template

File: src/app/components/course-item/course-item.component.ts (hunk 11 above)
Comment (titoffag) [reply src/app/components/course-item/course-item.component.ts]: I choose option - emit id as string

File: src/app/components/logo/logo.component.scss (hunk 18 above)
Comment (titoffag) [reply src/app/components/logo/logo.component.scss]: Yes, I'm. I import colors from this common style file. I can add separate style file called colors.scss for colors

File: src/index.html (hunk 24 above)
Comment (titoffag) [reply src/index.html]: what's the difference? just another option?

File: src/app/components/breadcrumbs/breadcrumbs.component.ts (hunk 6 above)
Comment (vitaliy-bobrov) [reply src/app/components/breadcrumbs/breadcrumbs.component.ts]: constructor?

File: src/app/components/course-item/course-item.component.scss [hunk 25]
```diff
@@ -0,0 +1,31 @@
+@import 'src/styles/colors.scss';
+@import 'src/styles/common.scss';
+
+:host li {
```
Comment (vitaliy-bobrov) [inline src/app/components/course-item/course-item.component.scss]: you can use nesting in SASS ;)

File: src/app/components/courses-list/courses-list.component.html [hunk 26]
```diff
@@ -1 +1,8 @@
-<p>courses-list works!</p>
+<ul>
+  <app-course-item *ngFor="let course of courses; trackBy: course ? course.id : null" [course]="course" (deletedCourse)="onDeletedCourse($event)">
```
Comment (vitaliy-bobrov) [inline src/app/components/courses-list/courses-list.component.html]: 1. move trackBy helper to method
2. Line is too long - could be formatted (maybe think about prettier setup for that https://github.com/schuchard/prettier-schematic)

File: src/app/components/logo/logo.component.html [hunk 27]
```diff
@@ -1 +1,9 @@
-<p>logo works!</p>
+<span>
```
Comment (vitaliy-bobrov) [inline src/app/components/logo/logo.component.html]: what that span is used for? remember that you already have a wrapper - custom element `app-logo`

File: src/app/components/panel/panel.component.scss [hunk 28]
```diff
@@ -0,0 +1,21 @@
+@import 'src/styles/colors.scss';
```
Comment (vitaliy-bobrov) [inline src/app/components/panel/panel.component.scss]: I believe you could omit extension in imports

**Pull request merged**